cogs/StaffRequirements.py

```diff
@@ -18,58 +28,93 @@ def __init__(self, bot: commands.Bot) -> None:
-                    True,
-                ],
+        )
+        time_since_join = datetime.now(tz=account.joined_at.tzinfo) - account.joined_at
+
+        fields = [
+            [
+                "Created",
+                f"{account.created_at.strftime('%B %d, %Y')}\n{strfdelta(time_since_creation.days/365.25,'year') if time_since_creation.days/365.25 >= 1 else ''} {strfdelta(time_since_creation.days % 365.25, 'day')} ago",
+                True,
             ],
-        ).build()
-        await ctx.respond(embed=embed, ephemeral=True)
+            [
+                "Joined",
+                f"{account.joined_at.strftime('%B %d, %Y')}\n{strfdelta(time_since_join.days/365.25, 'year') if time_since_join.days/365.25 >= 1 else ''} {strfdelta(time_since_join.days % 365.25, 'day')} ago",
+                True,
+            ],
+        ]
 
-        if (created_years_ago >= 1) and (joined_days_ago >= 30 or joined_years_ago >= 1):
+        # if staff requirements are met
+        # time since creation is at least 1 year (52 weeks) AND
+        # time since join is at least 30 days
+        if time_since_creation >= timedelta(weeks=52) and time_since_join >= timedelta(
+            days=30
+        ):
             embed = EmbedBuilder(
-                title="Application Link",
-                description=f"Congratulations, {account.mention}! You meet the basic requirements for staff. Please fill out the application form [here]({getenv('staff_google_form')}).",
+                title="Congratulations!",
+                description="You meet the basic requirements for staff. You may apply for staff below.",
+                fields=fields,
+                color=0x39FF14,  # GREEN
             ).build()
-            await ctx.respond(embed=embed, ephemeral=True)
+            await ctx.respond(embed=embed, ephemeral=True, view=StaffAppView())
         else:
             embed = EmbedBuilder(
                 title="Missing Requirements",
-                description=f"Unfortunately, you do not meet the basic requirements in order to apply for staff. Your account must be at least 1 year old and you must have been a member of the server for at least 30 days.",
+                description=f"Unfortunately, you do not meet the basic requirements in order to apply for staff.\n\nYour account must be at least 1 year old and you must have been a member of the server for at least 30 days.\n\nYou will be eligible to apply for staff in **{check_missing_requirements(time_since_creation, time_since_join)}**.",
+                fields=fields,
+                color=0xFF0000,  # RED
             ).build()
             await ctx.respond(embed=embed, ephemeral=True)
 
         log(f"User {account} checked their requirements in {ctx.guild}.")
 
 
-def setup(bot) -> None:
+def strfdelta(num: float, word: str) -> str:
```

Comment:
Hey by the way, you can simplify this code by just using the `humanize` module, which we already have installed in this repo.

For example
```py
import humanize
from datetime import timedelta

time_since_join = timedelta(days = 1023, hours = 3, minutes = 35, seconds = 12, microseconds = 827)
print(humanize.precisedelta(time_since_join, minimum_unit='hours'))
```

```
2 years, 9 months, 18 days and 3.59 hours
```